simple_app/app_simple.py

```diff
@@ -107,8 +114,19 @@ def listconcat(self, key: str, rtnmk: bool = True) -> Union[pd.DataFrame, str]:
         obj: List[dict] = self.results[key]  # the value for the given key
         df: pd.DataFrame = pd.concat([pd.DataFrame(objrow, index=[i])  # create dataframe from found dict
                                       for i, objrow in enumerate(obj)], ignore_index=True)  # every dict in the list
+        urlinks = []
+        if rtnmk and key == 'Spectra':
+            for src in df.spectrum.values:  # over every source in table
+                urllnk = src  # convert object name to url safe
+                srclnk = f'<a href="{urllnk}" target="_blank">Link</a>'  # construct hyperlink
+                urlinks.append(srclnk)  # add that to list
```

Comment:
There are ways to dynamically creating a new column in pandas without having to loop over every source. I believe this is using `df.apply(func)` where `func` is a function you define, though it may also work with a lambda function.
Anyways, this works for now, but just something to consider for the future.
Also you have `urllnk = src` with a comment that says that makes it url safe, but I don't see how that can be- it's just a variable rename.